src/cpp/library/queue

```diff
@@ -93,13 +99,19 @@ public:
   }
 };
 
-template <class T, class Container = deque<T>>
+template <class T, class Container = deque<T> >
 class queue
 {
   T buf[QUEUE_CAPACITY];
-  int _size = 0;
-  int head = 0;
-  int tail = 0;
+#if __cplusplus >= 201103L // Check if C++11 or later is being used
+  size_t _size = 0;
+  size_t head = 0;
+  size_t tail = 0;
+#else  
+  size_t _size;
+  size_t head;
+  size_t tail;
+#endif
```

Comment:
Can't this be unified? The class constructor already initializes those values to 0

Reply:
This is what I mentioned before :-)

See my comments: "The constructor method already initialises these attributes."

However, @mikhailramalho and @fbrausse had another suggestion :-)